compiler/sigmatch.nim

```diff
@@ -2213,6 +2220,10 @@ proc paramTypesMatchAux(m: var TCandidate, f, a: PType,
   of isNone:
     # do not do this in ``typeRel`` as it then can't infer T in ``ref T``:
     if a.kind in {tyProxy, tyUnknown}:
+      if a.kind == tyUnknown and c.inGenericContext > 0:
+        # don't bother with fauxMatch mechanism in generic type,
+        # reject match, typechecking will be delayed to instantiation
+        return nil
```

Comment:
Since typechecking will be delayed to instantiation, why is this code even called with `c.inGenericContext > 0`?

Reply:
ok